src/app.ts

```diff
@@ -1,9 +1,41 @@
-import express, { Request, Response } from 'express';
+import express, { Request, Response, NextFunction } from 'express';
+import { getFromCache, setToCache } from './services/cache';
 
 const app = express();
 
-app.use('/', (req: Request, res: Response) => {
-  res.send('Working');
+// Middleware function to handle caching
+const cacheMiddleware = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const cacheKey = req.originalUrl;
+    const cachedData = await getFromCache(cacheKey);
+
+    if (cachedData !== null) {
+      console.log('Retrieved data from Redis cache:', cachedData);
+      return res.send(cachedData);
+    }
+
+    // If data is not found in cache, continue to the next middleware/route
+    return next();
+  } catch (error) {
+    console.error('Error accessing Redis cache:', error);
+    return res.status(500).send('Internal Server Error');
+  }
+};
```

Comment:
Add cache middleware

Reply:
usually we don't use 500, it is for real disaster happens.